include/neml2/models/DataStore.h

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023, UChicago Argonne, LLC
+// All Rights Reserved
+// Software Name: NEML2 -- the New Engineering material Model Library, version 2
+// By: Argonne National Laboratory
+// OPEN SOURCE LICENSE (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#pragma once
+
+#include "neml2/base/NEML2Object.h"
+
+#include "neml2/base/Factory.h"
+#include "neml2/models/ContainsBuffers.h"
+
+namespace neml2
+{
+
+/// @brief  Parent class for all models that store data and can be sent with .to
+class DataStore : public NEML2Object
```

Comment:
```suggestion
class Data : public ContainsBuffers<NEML2Object>
```
I think this will address my question on slack.

Reply:
This won't work, we need to be able to access `Base::xxx` methods here.  This class can't do that unless it inherits from `Base`, just templating is not enough.

Reply:
I'm not sure I completely follow. Doesn't `ComtainsBuffers<NEML2Object>` inherit from `NEML2Object`?